src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,302 @@
+        String welcomeMessage = "Hello! I'm Duke\n" +
+                "What can I do for you?";
+
+        System.out.println("Hello from\n" + logo);
+
+        printWithTemplate(welcomeMessage);
+    }
+
+    public static void printByeMessage() {
+        String byeMessage = "Bye. Hope to see you again soon!";
+
+        printWithTemplate(byeMessage);
+    }
+
+    public static void saveFile() throws DukeException {
+
+        try {
+
+            FileWriter fw = new FileWriter(FILE_PATH.toString());
+
+            String textOut = "";
+
+            for (int i = 0; i < taskCount; i++) {
+                Task t = tasks.get(i);
+                textOut = textOut.concat(t.encrypt());
+                textOut = textOut.concat("\n");
+            }
+
+            fw.write(textOut);
+            fw.close();
+
+        } catch (IOException e) {
+            throw new DukeException("IO exception");
+        }
+    }
+
+    public static void createFile() throws IOException {
+
+        // Code referenced from
+        // https://stackoverflow.com/questions/2833853/create-whole-path-automatically-when-writing-to-a-new-file
+        Files.createDirectories(FILE_PATH.getParent());
+        Files.createFile(FILE_PATH);
+        printWithTemplate("New storage file created.");
+    }
+
+    public static void readFile() throws DukeException {
+        ArrayList<Task> parsedTasks = new ArrayList<>(100);
+        int parsedTaskCount = 0;
+
+        try {
+            File f = new File(FILE_PATH.toString());
+            Scanner s = new Scanner(f);
+
+            while (s.hasNext()) {
+                String line = s.nextLine();
+                String[] inputArray = line.split(" [ | ] ");
+                Task t;
+
+                /*
+                  type | isDone | description | attribute
+                  The input array splits the encrypted input into separate String values
+                 */
+
+                switch (inputArray[0]) {
+                case "T":
+                    t = new ToDo(inputArray[2]);
+                    break;
+                case "D":
+                    t = new Deadline(inputArray[2], inputArray[3]);
+                    break;
+                case "E":
+                    t = new Event(inputArray[2], inputArray[3]);
+                    break;
+                default:
+                    throw new DukeException("Parse Error");
+                }
+
+                if (inputArray[1].equals("1")) {
+                    t.setDone();
+                }
+
+                parsedTasks.add(parsedTaskCount, t);
+                parsedTaskCount++;
+            }
+
+            tasks = parsedTasks;
+            taskCount = parsedTaskCount;
+
+        } catch (FileNotFoundException e) {
+            try {
+                createFile();
+            } catch (IOException k) {
+                throw new DukeException(k.toString() + "\nProblem with reading the storage file.");
+            }
+        }
+
+    }
```

Comment:
Try to refactor this method into bunch of smaller methods (as a general rule no method should be more than 30 lines)